### PR TITLE
OCPBUGS-26602: Enable kubelet ECR authentiation

### PR DESCRIPTION
**Pull request by sebsoto**
Fixes a bug where authentication with ECR registries through the kubelet was no longer possible.

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -63,6 +63,11 @@ WORKDIR /build/windows-machine-config-operator/cloud-provider-azure/
 COPY cloud-provider-azure/ .
 RUN GOOS=windows go build -o azure-cloud-node-manager.exe ./cmd/cloud-node-manager
 
+# Build ecr-credential-provider
```
Comment (jrvaldes) [inline build/Dockerfile:66]: Changes may be required in the [mid-stream dockerfile](https://gitlab.cee.redhat.com/openshift-winc-midstream/openshift-winc-midstream/-/blob/rhaos-4.16-rhel-9/distgit/containers/windows-machine-config-operator/Dockerfile?ref_type=heads#L1) as well.

File: pkg/services/services.go [hunk 2]
```diff
@@ -241,6 +241,8 @@ func generateKubeletArgs(argsFromIgnition map[string]string, debug bool) ([]stri
 		// Windows host.
 		// See: https://kubernetes.io/docs/concepts/configuration/windows-resource-management/#resource-management-cpu
 		"--windows-priorityclass=" + windowsPriorityClass,
+		"--image-credential-provider-bin-dir=" + windows.K8sDir,
```
Comment (jrvaldes) [inline pkg/services/services.go:244]: consider using the KubeletConfiguration instead

https://github.com/openshift/windows-machine-config-operator/blob/6ef54058a437967ca61755c31631898679a83a20/pkg/nodeconfig/nodeconfig.go#L662

**Review by jrvaldes — COMMENTED** [review]
@sebsoto  thanks for putting this  together, Mostly LGTM. Please review my comments. 

It feels like too many changes/refactoring for a bug fix.

File: pkg/windows/windows.go [hunk 3]
```diff
@@ -201,7 +205,13 @@ func getFilesToTransfer(platform *config.PlatformType) map[string]string {
 		payload.NetworkConfigurationScript:     remoteDir,
 	}
 
-	if platform != nil && *platform == config.AzurePlatformType {
+	if platform == nil {
+		return srcDestPairs
+	}
```
[the remark this answers is not in the document]
Comment (jrvaldes) [reply pkg/windows/windows.go:210]: True

File: build/Dockerfile [hunk 4]
```diff
@@ -63,6 +63,11 @@ WORKDIR /build/windows-machine-config-operator/cloud-provider-azure/
...
+WORKDIR /build/windows-machine-config-operator/cloud-provider-aws/
+COPY cloud-provider-aws/ .
+RUN env -u VERSION GOOS=windows make ecr-credential-provider
```
Comment (mansikulkarni96) [inline build/Dockerfile:69]: Have we explored the cross-build make target, does it help us with getting the extensions we need for Windows? https://github.com/openshift/cloud-provider-aws/blob/51180c32ceb70298faad1aac1022d8983fdf2e78/Makefile#L208

File: pkg/services/services.go (hunk 2 above)
Comment (sebsoto) [reply pkg/services/services.go:244]: I was not able to find this flag in that struct

File: pkg/nodeconfig/nodeconfig.go [hunk 5]
```diff
@@ -23,10 +24,12 @@ import (
 	cloudproviderapi "k8s.io/cloud-provider/api"
 	cloudnodeutil "k8s.io/cloud-provider/node/helpers"
 	"k8s.io/kubectl/pkg/drain"
+	kubeletconfigv1 "k8s.io/kubelet/config/v1"
 	kubeletconfig "k8s.io/kubelet/config/v1beta1"
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply pkg/nodeconfig/nodeconfig.go:28]: Doing so may cause unexpected behavior changes which I wish to avoid

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Comment by mansikulkarni96** [comment]
/approve
Thanks for fixing this!

**Pull request merged**